ccvm_simulators/solvers/pumped_langevin_solver.py

```diff
@@ -0,0 +1,637 @@
+        )
+
+        alpha = hyperparameters["alpha"]
+        beta1 = hyperparameters["beta1"]
+        beta2 = hyperparameters["beta2"]
+        epsilon = 1e-8
+
+        # Choose desired update method.
+        if hyperparameters["add_assign"]:
+            update_grads_with_moment2 = (
+                lambda grads, mhat, vhat: grads
+                + alpha * torch.div(mhat, torch.sqrt(vhat) + epsilon)
+            )
+
+            update_grads_without_moment2 = lambda grads, mhat: grads + alpha * mhat_c
+        else:
+            update_grads_with_moment2 = lambda grads, mhat, vhat: alpha * torch.div(
+                mhat, torch.sqrt(vhat) + epsilon
+            )
+
+            update_grads_without_moment2 = lambda grads, mhat: alpha * mhat_c
+
+        # Initialize first moment vector
+        m_c = torch.zeros((batch_size, problem_size), dtype=torch.float, device=device)
+        # Initialize second moment conditionally
+        if not beta2 == 1.0:
+            v_c = torch.zeros(
+                (batch_size, problem_size), dtype=torch.float, device=device
+            )
+        else:
+            v_c = None
+
+        # Define a method conditionally for estimating instantaneous pump rate
+        if pump_rate_flag:
+            pump_field = lambda n: pump * (n + 1) / iterations
+        else:
+            pump_field = lambda n: pump
+
+        # Perform the solve with Adam over the specified number of iterations
+        for i in range(iterations):
+            # Calculate gradient
+            c_grads = self.calculate_grads(c, S)
+
+            # Update biased first moment estimate
+            m_c = beta1 * m_c + (1.0 - beta1) * c_grads
+
+            # Compute bias correction in 1st moment
+            beta1i = 1.0 - beta1 ** (i + 1)
+            mhat_c = m_c / beta1i
+
+            # Conditional second moment estimation
+            if not beta2 == 1.0:
+                # Update biased 2nd moment estimate
+                v_c = beta2 * v_c + (1.0 - beta2) * torch.pow(c_grads, 2)
+
+                # Compute bias correction in 2nd moment
+                beta2i = 1.0 - beta2 ** (i + 1)
+                vhat_c = v_c / beta2i
+
+                # Compute bias corrected grad using 1st and 2nd moments
+                # in the form of element-wise division
+                c_grads = update_grads_with_moment2(c_grads, mhat_c, vhat_c)
+            else:
+                # Compute bias corrected grad only with 1st moment
+                c_grads = update_grads_without_moment2(c_grads, mhat_c)
+
+            wiener_increment_c = wiener_dist_c.sample((problem_size,)).transpose(
+                0, 1
+            ) * np.sqrt(dt)
+
+            # Calculate contribution of pumped field
+            c_pow = torch.pow(c, 2)
+            c_pump = torch.einsum("cj,cj -> cj", -1 + pump_field(i) - c_pow, c)
+
+            # Update variable
+            c += dt * feedback_scale * (c_pump + c_grads) + sigma * wiener_increment_c
```

Comment:
Same comment here as I wrote above about the `feedback_scale` in front of the gradient term instead of the whole drift term. 
```suggestion
            c += dt * (c_pump +  feedback_scale * c_grads) + sigma * wiener_increment_c
```